### PR TITLE
Support for shared key keyvault url in LA

### PR DESCRIPTION
**Pull request by juhi037**
## Description
</br>
 Brief description of the requested change.

## Checklist
- [ ] I have read the instructions mentioned in [Contribute to Code](/Contributing.md).
- [ ] I have read and understood the criteria described under [submitting changes](/Contributing.md#submitting-changes). 
- [ ] The title of the PR clearly describes the intent of the PR.
- [ ] This PR does not introduce any breaking changes to the code.

### CODE REVIEW

File: src/AzSK.ADO/Framework/Helpers/Helpers.ps1 [hunk 1]
```diff
@@ -953,5 +953,42 @@ class Helpers {
 
         return $obj
     }
+
+    static [System.Security.SecureString] GetVariableFromKVUrl($url){
+        $variable = $null;
+        $Context = @(Get-AzContext -ErrorAction SilentlyContinue )
```
Comment (SaikumarRagala) [inline src/AzSK.ADO/Framework/Helpers/Helpers.ps1:959]: Lets maintain the naming consistent...  $Context --> $context

File: src/AzSK.ADO/AlertMonitoring/LogAnalytics.ps1 [hunk 2]
```diff
@@ -60,7 +60,18 @@ function Set-AzSKADOMonitoringSettings
         [Parameter(Mandatory = $true, HelpMessage="Use -Disable option to clean the Log Analytics setting under the current instance.", ParameterSetName = "Disable")]
         [switch]
 		[Alias("dsbl")]
-        $Disable
+        $Disable,
+
+		[Parameter(Mandatory = $false, HelpMessage="Key vault URL of the shared key of your Log Analytics instance.", ParameterSetName = "Setup")]
```
Comment (SaikumarRagala) [inline src/AzSK.ADO/AlertMonitoring/LogAnalytics.ps1:65]: Key vault URL that stores shared key

File: src/AzSK.ADO/AlertMonitoring/LogAnalytics.ps1 [hunk 3]
```diff
@@ -60,7 +60,18 @@ function Set-AzSKADOMonitoringSettings
...
+        [string]
+		[Alias("wkeyurl", "SharedKeyUrl")]
+        $LAWSSharedKeyUrl,
+
+		[Parameter(Mandatory = $false, HelpMessage="Key vault URL of the shared key of your alternate Log Analytics instance.", ParameterSetName = "Setup")]
```
Comment (SaikumarRagala) [inline src/AzSK.ADO/AlertMonitoring/LogAnalytics.ps1:70]: Same as above

File: src/AzSK.ADO/AlertMonitoring/LogAnalytics.ps1 [hunk 4]
```diff
@@ -80,6 +91,22 @@ function Set-AzSKADOMonitoringSettings
 					$appSettings.LAWSId = $LAWSId
 					$appSettings.LAWSSharedKey = $LAWSSharedKey
 				}
+				elseif(-not [string]::IsNullOrWhiteSpace($LAWSSharedKeyUrl) ){
+					if([string]::IsNullOrWhiteSpace($LAWSId)){
+						[EventBase]::PublishGenericCustomMessage("You need to send both the LAWSId and LAWSSharedKeyUrl", [MessageType]::Error);
```
Comment (SaikumarRagala) [inline src/AzSK.ADO/AlertMonitoring/LogAnalytics.ps1:96]: Both the parameters LAWSId and LAWSSharedKeyUrl are required

File: src/AzSK.ADO/AlertMonitoring/LogAnalytics.ps1 [hunk 5]
```diff
@@ -91,6 +118,22 @@ function Set-AzSKADOMonitoringSettings
 					$appSettings.AltLAWSId = $AltLAWSId
 					$appSettings.AltLAWSSharedKey = $AltLAWSSharedKey
 				}
+				elseif(-not [string]::IsNullOrWhiteSpace($AltLAWSSharedKeyUrl) ){	
+					if([string]::IsNullOrWhiteSpace($AltLAWSId)){
+						[EventBase]::PublishGenericCustomMessage("You need to send both the AltLAWSId and AltLAWSSharedKeyUrl", [MessageType]::Error);
```
Comment (SaikumarRagala) [inline src/AzSK.ADO/AlertMonitoring/LogAnalytics.ps1:123]: Same as above

**Review by Garima-msft — APPROVED** [review]

**Review by SaikumarRagala — APPROVED** [review]

**Pull request merged**